mediagrains/grain.py

```diff
@@ -1223,6 +1234,9 @@ def __init__(self, meta, data):
                 self.meta['grain']['cog_audio'][key] = 0
         self.meta['grain']['cog_audio']['format'] = int(self.meta['grain']['cog_audio']['format'])
 
+    def final_origin_timestamp(self):
+        return (self.origin_timestamp + TimeOffset.from_count(self.samples - 1, 1, self.sample_rate))
```

Comment:
swap the rate params, ie. `TimeOffset.from_count(self.samples - 1, self.sample_rate, 1)`